optimism/src/main.rs

```diff
@@ -88,17 +98,37 @@ pub fn main() -> ExitCode {
     // Initialize the environments
     // The Keccak environment is extracted inside the loop
     let mut mips_wit_env = mips_witness::Env::<Fp>::create(cannon::PAGE_SIZE as usize, state, po);
-    let mut mips_con_env = mips_constraints::Env::<Fp> {
-        scratch_state_idx: 0,
-        constraints: Vec::new(),
-        lookups: Vec::new(),
-    };
+    let mut mips_con_env = mips_constraints::Env::<Fp>::default();
     // The keccak environment is extracted inside the loop
 
     // Initialize the circuits. Includes pre-folding witnesses.
     let mut mips_trace = MIPSTrace::<Fp>::new(DOMAIN_SIZE, &mut mips_con_env);
     let mut keccak_trace = KeccakTrace::<Fp>::new(DOMAIN_SIZE, &mut KeccakEnv::<Fp>::default());
 
+    let _mips_folding = {
```

Comment:
I can move this in a draft PR the reviewers do not accept this change. This change shows that the MIPS instructions can now be used with the `DecomposableFoldingScheme`, but it is not in a final state.